web/src/muted_topics_ui.js

```diff
@@ -29,6 +30,7 @@ export function rerender_for_muted_topic(old_muted_topics) {
     for (const topic_data of maybe_affected_topics) {
         recent_topics_ui.update_topic_is_muted(topic_data.stream_id, topic_data.topic);
     }
+    message_util.render_old_messages_for_stream_and_topic_unmuting();
```

Comment:
This call is not required for `unmuting topics` in `unmuted streams`, but only in `muted streams`. So, calling this function for `unmuting topics` within `unmuted streams` is not necessary. However, https://github.com/zulip/zulip/pull/25151 has implemented two separate functions for `unmuting topics` within muted and unmuted streams, after which we can just place this call for the muted stream function, which will optimize the code further.